src/components/contextual/pages/pools/StakedPoolsTable.vue

```diff
@@ -0,0 +1,192 @@
+  gaugeShares: {
+    __args: {
+      where: { user: account.value.toLowerCase() }
+    },
+    balance: 1,
+    gauge: {
+      poolId: 1
+    }
+  },
+  liquidityGauges: {
+    __args: {
+      where: {
+        poolId_in: userPoolsAddresses.value
+      }
+    }
+  }
+}));
+
+const { data: stakedPoolsRes, isLoading: isLoadingPools } = usePoolsQuery(
+  ref([]),
+  {},
+  {
+    poolIds: userStakedPools
+  }
+);
+
+/** COMPUTED */
+const stakedPools = computed(() => stakedPoolsRes.value?.pages[0].pools);
+
+// a map of poolId-stakedBPT for the connected user
+const stakedBalanceMap = computed(() => {
+  const map: Record<string, string> = {};
+  if (!gaugeSharesRes.value) return map;
+  for (const gaugeShare of gaugeSharesRes.value?.gaugeShares) {
+    map[gaugeShare.gauge.poolId] = gaugeShare.balance;
+  }
+  return map;
+});
+
+// first retrieve all the pools the user has liquidity for
+const { data: userPools } = useUserPoolsQuery();
+
+// if the staked pool (from gauge subgraph) is not in the
+// user pools response, stitch it together. This is because
+// when all the BPT for a pool is staked, then it will not
+// show up in the user pools query
+const allStakedPools = computed(() => {
+  const userPoolIds = userPools.value?.pools.map(pool => pool.id);
+  const stakedPoolIds = stakedPools.value?.map(pool => pool.id);
+
+  // first find the staked pools which are not in the user pools
+  const maxStakedPools = (stakedPools.value || [])
+    .filter(pool => {
+      return !userPoolIds?.includes(pool.id);
+    })
+    .map(pool => ({
+      // then indicate that those pools are maximal staked with a variable
+      ...pool,
+      stakedPct: '1'
+    }));
+
+  // now get the pools which are both staked, but also have BPT available for staking
+  // NOTE: we are iterating through user pools here so we can access the bpt var
+  const partiallyStakedPools = (userPools.value?.pools || [])
+    .filter(pool => {
+      return stakedPoolIds?.includes(pool.id);
+    })
+    .map(pool => {
+      // calculate the staked percentage by using the staked balance
+      // pulled from the gauge subgraph
+      const stakedBalance = stakedBalanceMap.value[pool.id];
+      const unstakedBalance = pool.bpt;
+      const stakedPct = bnum(stakedBalance).div(
+        bnum(stakedBalance).plus(unstakedBalance)
+      );
+      return {
+        ...pool,
+        stakedPct: stakedPct.toString()
+      };
+    });
+
+  // also include any pools where there is no staked BPT at all
+  const availableGaugePoolIds = (
+    gaugeSharesRes.value?.liquidityGauges || []
+  ).map(gauge => gauge.poolId);
+  const stakablePools = (userPools.value?.pools || [])
+    .filter(pool => {
+      return availableGaugePoolIds?.includes(pool.id);
+    })
+    .map(pool => ({
+      ...pool,
+      stakedPct: '0'
+    }));
+
+  // now mash them together
+  return [...stakablePools, ...partiallyStakedPools, ...maxStakedPools];
+});
```

Comment:
Not a massive fan of huge computed properties like this. Feel like it should be broken up into 4 computed properties: stakablePools, partiallyStakedPools, partiallyStakedPools and then allStakedPools is just the combination of the other 3.